scripts/components/api-changes-validator/api_usage_generator.ts

```diff
@@ -0,0 +1,81 @@
+import ts from 'typescript';
+import { EOL } from 'os';
+import { UsageStatements } from './types.js';
+import {
+  ClassUsageStatementsGenerator,
+  EnumUsageStatementsGenerator,
+  ImportUsageStatementsGenerator,
+  TypeUsageStatementsGenerator,
+  VariableUsageStatementsGenerator,
+} from './api_usage_statements_generators.js';
+
+/**
+ * Generates API usage using API.md definition.
+ */
+export class ApiUsageGenerator {
+  /**
+   * Creates generator.
+   */
+  constructor(
+    private readonly packageName: string,
+    private readonly apiReportAST: ts.SourceFile
+  ) {}
+
+  generate = (): string => {
+    const importStatements: Array<string> = [];
+    const usageStatements: Array<string> = [];
+
+    // go over top level statements and generate usage for them
+    for (const statement of this.apiReportAST.statements) {
+      const statementsForNode = this.generateStatementsForNode(statement);
+      if (statementsForNode) {
+        if (statementsForNode.importStatement) {
+          importStatements.push(statementsForNode.importStatement);
+        }
+        if (statementsForNode.usageStatement) {
+          usageStatements.push(statementsForNode.usageStatement);
+        }
+      }
+    }
+
+    return `${importStatements.join(EOL)}${EOL}${EOL}${usageStatements.join(
+      EOL
+    )}${EOL}`;
+  };
+
+  private generateStatementsForNode = (
+    node: ts.Node
+  ): UsageStatements | undefined => {
+    if (node.kind === ts.SyntaxKind.ImportDeclaration) {
```

Comment:
nit: should we use `switch` instead?

Reply:
we could do that, the reason `if` remained here is that these blocks were much heavier at some point and linter does not allow to define new `const/var` inside switch blocks.